### PR TITLE
feat(OpenAI Node): Overhaul

### PR DESCRIPTION
**Pull request by michael-radency**
## Summary
Update OpenAI node

Small tweaks on custom code tool and other

Other OpenAI node tickets 

## Related tickets and issues
https://linear.app/n8n/issue/NODE-976/ai-node-tweaks-and-openai-node-update

### CODE REVIEW

**Comment by BleedingDev** [comment]
Is there something I could do for this PR? I'd like to contribute to make it pass sooner. :)

File: packages/@n8n/nodes-langchain/nodes/vendors/OpenAi/actions/image/generate.operation.ts [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+				],
+				displayOptions: {
+					show: {
+						'/model': ['dall-e-3'],
+					},
+				},
+				default: 'vivid',
+			},
+			{
+				displayName: 'Responde with Image URL(s)',
```
Comment (OlegIvaniv) [inline packages/@n8n/nodes-langchain/nodes/vendors/OpenAi/actions/image/generate.operation.ts:160]: Typo: `Responde`

File: packages/@n8n/nodes-langchain/nodes/vendors/OpenAi/actions/text/classify.operation.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		type: 'collection',
+		default: {},
+		options: [
+			{
+				displayName: 'Use Stable Model',
+				name: 'useStableModel',
+				type: 'boolean',
+				default: false,
+				description:
+					'Whether to use the stable version of the model instead of the latest version, accuracy may be sligthly lower',
```
Comment (OlegIvaniv) [inline packages/@n8n/nodes-langchain/nodes/vendors/OpenAi/actions/text/classify.operation.ts:37]: Typo: `sligthly` -> `slightly`

File: packages/@n8n/nodes-langchain/nodes/vendors/OpenAi/actions/text/message.operation.ts [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+							{
+								name: 'Assistant',
+								value: 'assistant',
+								description: 'Tell the model to adopt a specific tone or personality',
+							},
+							{
+								name: 'System',
+								value: 'system',
+								description:
+									"Usualy used to set the model's behavior or context for the next user message",
```
Comment (OlegIvaniv) [inline packages/@n8n/nodes-langchain/nodes/vendors/OpenAi/actions/text/message.operation.ts:63]: Typo: `Usualy` -> `Usually`

**Review by OlegIvaniv — CHANGES_REQUESTED** [review]
Really nice job on this one! 👏 
I love that you've added tests and `updateDisplayOptions` helper also seems quite helpful. 
I left just a few comments with typos, btw if you're using VS Code I would really recommend to install [Code Spell Checker extension](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker), it saved me many times :D

**Review by OlegIvaniv — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.29.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.29.0)

**Pull request merged**